lib/sedna/service/client.py

```diff
@@ -0,0 +1,218 @@
+
+        try:
+            conn = websockets.connect(
+                self.uri, **self.kwargs
+            )
+            self.ws = await conn.__aenter__()
+            await self.ws.send(json.dumps({'type': 'subscribe', 'client_id': self.client_id}))
+
+            res = await self.ws.recv()
+            return res
+        except ConnectionRefusedError:
+            sednaLogger.info(f"{self.uri} connection was refused by server")
+            raise
+        except ConnectionClosedError:
+            sednaLogger.info(f"{self.uri}  connection lost")
+            raise
+        except ConnectionClosedOK:
+            sednaLogger.info(f"{self.uri}  connection closed")
+            raise
+        except InvalidStatusCode as err:
+            sednaLogger.info(f"{self.uri}  Websocket failed - with invalid status code {err.status_code}")
+            raise
+        except WebSocketException as err:
+            sednaLogger.info(f"{self.uri}  Websocket failed - with {err}")
+            raise
+        except OSError as err:
+            sednaLogger.info(f"{self.uri} Connection failed - with {err}")
+            raise
+        except Exception:
+            sednaLogger.exception(f"{self.uri} Websocket Error")
+            raise
+
+    async def _send(self, data):
+        error = ""
+        for _ in range(self._retry):
+            try:
+                await self.ws.send(data)
+                result = await self.ws.recv()
+                return result
+            except Exception as e:
+                error = e
+                sednaLogger.warning(f"send data error: {error}")
+                time.sleep(self._retry_interval_seconds)
+        sednaLogger.error(f"websocket error: {error}, retry times: {self._retry}")
+        return None
+
+    def send(self, data, msg_type="message", job_name=""):
+        loop = asyncio.get_event_loop()
+        j = json.dumps({
+            "type": msg_type, "client": self.client_id,
+            "data": data, "job_name": job_name
+        })
+        data_json = loop.run_until_complete(self._send(j))
+        if data_json is None:
+            sednaLogger.error("send data to agg worker failed, exist worker")
+            return
+        res = json.loads(data_json)
+        return res
+
+
+class ModelClient:
+    """Remote model service"""
+
+    def __init__(self, service_name, version="", host="127.0.0.1", port="8080", protocol="http"):
+        self.server_name = f"{service_name}{version}"
+        self.endpoint = f"{protocol}://{host}:{port}/{service_name}"
+
+    def check_server_status(self):
+        try:
+            http_request(url=self.endpoint, method="GET", json={})
+        except:
+            return False
+        else:
+            return True
+
+    def inference(self, x, **kwargs):
+        """Use the remote big model server to inference."""
+        json_data = deepcopy(kwargs)
+        json_data.update({"data": x})
+        _url = "{}/{}".format(self.endpoint, "predict")
```

Comment:
this _url is fixed, can be initialized once

Reply:
fix